src/Cake/Commands/DebugCommand.cs

```diff
@@ -38,15 +38,25 @@ public bool Execute(CakeOptions options)
                 throw new ArgumentNullException("options");
             }
 
-            var message = "Attach debugger to process {0} to continue";
+            var message = "Attach debugger to process {0} to continue or press <ESC> to skip and launch debugger upon first #break-directive.";
             var pid = _debugger.GetProcessId();
 
             _log.Debug("Performing debug...");
             _log.Information(Verbosity.Quiet, message, pid);
 
-            _debugger.WaitForAttach(Timeout.InfiniteTimeSpan);
-
-            _log.Debug("Debugger attached");
+            while (true)
+            {
+                if (Console.KeyAvailable && Console.ReadKey(true).Key == ConsoleKey.Escape)
```

Comment:
@patriksvensson Should this maybe be abstracted? I mean add **KeyAvailable** + **ReadKey** to **IConsole**, then it would be possible to write unit test around this also...